src/NativeAdComponents.js

```diff
@@ -146,6 +146,61 @@ export const MediaView = (props) => {
   if (!nativeAdView) return null;
 
   return (
-    <View {...props} ref={viewRef}/>
+    <View {...props} ref={viewRef} />
+  );
+};
+
+export const StarRatingView = (props) => {
+  const {style, ...otherProps} = props;
+
+  const maxStarCount = 5;
+  const starColor = style.color ?? "#ffe234";
+  const starSize = style.fontSize ?? 10;
+
+  const {nativeAd, nativeAdView} = useContext(NativeAdViewContext);
+
+  const FilledStar = useCallback(({size,color}) => {
+    return (
+      // black star in unicode
+      <Text style={{fontSize: size, color: color}}>{String.fromCodePoint(0x2605)}</Text>
+    );
+  });
+
+  const EmptyStar = useCallback(({size,color}) => {
+    return (
+      // white star in unicode
+      <Text style={{fontSize: size, color: color}}>{String.fromCodePoint(0x2606)}</Text>
+    );
+  });
+
+  const StarView = useCallback(({index, rating, size, color}) => {
+    const width = (rating - index) * size;
+    return (
+      <View>
+        <EmptyStar size={size} color={color} />
+        {
+          (rating > index) &&
+            <View style={{ width: width, overflow: 'hidden', position: 'absolute'}}>
+              <FilledStar style={{top:0, left:0}} size={size} color={color} />
+            </View>
+        }
+      </View>
+    );
+  });
+
+  if (!nativeAdView) return null;
+
+  return (
+    <View style={[style, {flexDirection: 'row', alignItems: 'center'}]}>
+    {
+      Array(maxStarCount).fill().map((j, i) => {
```

Comment:
A couple nits:

- For `j` - since it is not used, and represents the element in the Array (which currently is undefined), we should just update to `_`.
- For `i` - I'd prefer a more readable var like `index`.

Reply:
Actually, it's kind of weird to use `Array(maxStarCount).fill().map` when we just need to iterate to 5. Could we try using a for-loop and populating (then returning) an empty array with the `StarView` and `Text` elements?

Reply:
Also, are we able to handle half stars? (e.g. 3.5)

Reply:
> Actually, it's kind of weird to use `Array(maxStarCount).fill().map` when we just need to iterate to 5. Could we try using a for-loop and populating (then returning) an empty array with the `StarView` and `Text` elements?

Do you want to do like this?
```
for ( let index = maxStarCount: index--; ) {
 ...
}
```

> Also, are we able to handle half stars? (e.g. 3.5)

Yes, it can.  The snapshot was attached.